jib-cli/src/test/java/com/google/cloud/tools/jib/cli/cli2/CacheDirectoriesTest.java

```diff
@@ -46,26 +47,38 @@ public void testCacheDirectories_defaults() {
             XdgDirectories.getCacheHome()
                 .resolve("cli")
                 .resolve("projects")
-                .resolve(
-                    CacheDirectories.getProjectCacheDirectoryFromProject(
-                        Paths.get("some-context"))));
+                .resolve(CacheDirectories.getProjectCacheDirectoryFromProject(buildContext)));
   }
 
   @Test
-  public void testCacheDirectories_configuredValuesIgnoresBuildContext() {
+  public void testCacheDirectories_configuredValuesIgnoresBuildContext() throws IOException {
     CommonCliOptions commonCliOptions =
         CommandLine.populateCommand(
             new CommonCliOptions(),
             "-t=ignored",
             "--base-image-cache=test-base-image-cache",
             "--project-cache=test-project-cache");
-    CacheDirectories cacheDirectories =
-        CacheDirectories.from(commonCliOptions, Paths.get("ignored"));
+    Path ignoredContext = temporaryFolder.newFolder("ignored").toPath();
+    CacheDirectories cacheDirectories = CacheDirectories.from(commonCliOptions, ignoredContext);
 
     assertThat(cacheDirectories.getBaseImageCache()).hasValue(Paths.get("test-base-image-cache"));
     assertThat(cacheDirectories.getProjectCache()).isEqualTo(Paths.get("test-project-cache"));
   }
 
+  @Test
+  public void testCacheDirectories_failIfContextIsNotDirectory() throws IOException {
+    Path badContext = temporaryFolder.newFile("a.file").toPath();
```

Comment:
Not a big deal, but `temporaryFolder.newFile().toPath()` will work too.

And use the new exception assertion style in jib-cli (using a lambda).